### ISSUE TITLE
False-positive for `empty_enum_arguments ` rule

### ISSUE THREAD

**Issue by umbertovolta** [realm/SwiftLint#3562]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `empty_enum_arguments` rule matches in `if` statements when using dot notation on a function without parameters.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ Pods/SwiftLint/swiftlint lint empty_enum_arguments.swift
Linting Swift files at paths empty_enum_arguments.swift
Linting 'empty_enum_arguments.swift' (1/1)
/Users/umv/dev/swiftlint_test/empty_enum_arguments.swift:13:24: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated values if they are not used. (empty_enum_arguments)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.43.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
CocoaPods
* Paste your configuration file:

```yml
only_rules:
    - empty_enum_arguments
```

* Which Xcode version are you using (check `xcodebuild -version`)?
12.4
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
struct TestSwiftlint {
  struct Person: Equatable {
    let name: String

    static func mocked(name: String = "Mock") -> Self {
      return Self(name: name)
    }
  }

  let pippo = Person(name: "Pippo")

  func check() {
    if pippo == .mocked() { // <-- empty_enum_arguments
      print("Pippo is mocked")
    } else {
      print("Pippo is real")
    }
  }
}
```
